File: Selenium/Activities/Activity21.java
package activity;

import java.time.Duration;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity21 {

	public static void main(String[] args) {

		WebDriver driver = new ChromeDriver();
		WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(5000));
		
		driver.get("https://training-support.net/webelements/tabs");
		
		System.out.println("Title of the page: "+ driver.getTitle());
		
		driver.findElement(By.cssSelector(".bg-purple-200.svelte-1bfwg86")).click();
		
		wait.until(ExpectedConditions.numberOfWindowsToBe(2));
		
		System.out.println("All open windows: " + driver.getWindowHandles());
		
		for(String window : driver.getWindowHandles()) {
			driver.switchTo().window(window);
		}
		
		System.out.println("New window's title: " + driver.getTitle());
		
		String message = driver.findElement(By.cssSelector(".mt-5")).getText();
		System.out.println("New window's messgage: " + message);
		
		driver.findElement(By.xpath("//button[text()='Open Another One']")).click();
		
		wait.until(ExpectedConditions.numberOfWindowsToBe(3));
		
		for(String window : driver.getWindowHandles()) {
			driver.switchTo().window(window);
		}
		
		System.out.println("Another new window's messgage: " + driver.findElement(By.cssSelector(".mt-5")).getText());
		
		driver.quit();

	}

}

File: Selenium/Activities/activity_21.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait

with webdriver.Firefox() as driver:

    wait = WebDriverWait(driver, timeout=5)

    driver.get("https://training-support.net/webelements/tabs")

    print("Title of the page: ", driver.title)

    driver.find_element(By.CSS_SELECTOR, ".bg-purple-200.svelte-1bfwg86").click()

    wait.until(EC.number_of_windows_to_be(2))

    print("All open windows: ", driver.window_handles)

    driver.switch_to.window(driver.window_handles[1])

    print("New window's title: " + driver.title)

    message = driver.find_element(By.CSS_SELECTOR, ".mt-5").text
    print("New window's messgage: ", message)

    driver.find_element(By.XPATH, "//button[text()='Open Another One']").click()
    
    wait.until(EC.number_of_windows_to_be(3))

    print("All open windows: ", driver.window_handles)

    driver.switch_to.window(driver.window_handles[2])

    print("Another new window's messgage: " + driver.find_element(By.CSS_SELECTOR, ".mt-5").text)
